lib/class-wp-json-posts-controller.php

```diff
@@ -106,4 +869,196 @@ public function prepare_item_for_response( $post, $request ) {
+					'type'         => 'string',
+					'format'       => 'date-time',
+				),
+			),
+		);
+
+		$post_type_obj = get_post_type_object( $this->post_type );
+		if ( $post_type_obj->hierarchical ) {
+			$schema['properties']['parent_id'] = array(
+				'description'      => 'The ID for the parent of the Post.',
+				'type'             => 'integer',
+				);
+		}
+
+		if ( 'post' === $this->post_type ) {
+			$schema['properties']['sticky'] = array(
+				'description'      => 'Whether or not the Post should be treated as sticky.',
+				'type'             => 'bool',
+				);
+		}
+
+		$post_type_attributes = array(
+			'title',
+			'editor',
+			'author',
+			'excerpt',
+			'thumbnail',
+			'comments',
+			'revisions',
+			'page-attributes',
+			'post-formats',
+			);
+		foreach( $post_type_attributes as $attribute ) {
+			if ( ! post_type_supports( $this->post_type, $attribute ) ) {
+				continue;
+			}
+
+			switch( $attribute ) {
+
+				case 'title':
+					$schema['properties']['title'] = array(
+						'description'     => 'The title for the Post.',
+						'type'            => 'string',
+						'required'        => true,
+						);
+					break;
+
+				case 'editor':
+					$schema['properties']['content'] = array(
+						'description'     => 'The content for the Post.',
+						'type'            => 'string',
+						'required'        => true,
+						);
+					break;
+
+				case 'author':
+					$schema['properties']['author_id'] = array(
+						'description'     => 'The ID for the author of the Post.',
+						'type'            => 'integer',
+						);
+
+				case 'excerpt':
+					$schema['properties']['excerpt'] = array(
+						'description'     => 'The excerpt for the Post.',
+						'type'            => 'string',
+						'required'        => true,
+						);
+					break;
+
+				case 'thumbnail':
+					$schema['properties']['featured_image_id'] = array(
+						'description'     => 'A featured image for the Post.',
+						'type'            => 'integer',
+						);
+					break;
+
+				case 'comments':
+					$schema['properties']['comment_status'] = array(
+						'description'     => 'Whether or not comments are open on the Post.',
+						'type'            => 'string',
+						'enum'            => array( 'open', 'closed' ),
+						);
+					$schema['properties']['ping_status'] = array(
+						'description'     => 'Whether or not the Post can be pinged.',
+						'type'            => 'string',
+						'enum'            => array( 'open', 'closed' ),
+						);
+					break;
+
+				case 'page-attributes':
+					$schema['properties']['menu_order'] = array(
+						'description'     => 'The order of the Post in relation to other Posts of its type.',
+						'type'            => 'integer',
+						);
+					break;
+
+				case 'post-formats':
+					$schema['properties']['format'] = array(
+						'description'     => 'The format for the post',
+						'type'            => 'string',
+						'enum'            => get_post_format_slugs(),
+						);
+					break;
+
+			}
+
+		}
+
```

Comment:
@danielbachhuber Why for Posts are you not providing a `required` array, like you do for Users and Terms?